web-wallet/src/lib/containers/MigrateContract/MigrateContract.svelte

```diff
@@ -1,133 +1,196 @@
+        migrationContract: "0x",
+        name: "ERC-20",
+      },
     },
-    eth: {
-      chainId: mainnet.id,
-      name: "ERC-20",
+    testnet: {
+      "BEP-20": {
+        chainId: sepolia.id,
+        contract: "0xC416f5d2AE6BAec2a23f412Df11166afC35CAba2",
+        migrationContract: "0x9f5d1c067710fc6ed49a6444afd69b64799a57b6",
+        name: "BEP-20",
+      },
+      "ERC-20": {
+        chainId: sepolia.id,
+        contract: "0x92DA9BE2039E818bB78223A6BA7C85CC2b17D8D5",
+        migrationContract: "0x63fd2B12034e108BCe73e8832b7dabC8bd67f738",
+        name: "ERC-20",
+      },
     },
   };
 
   const options = [
-    { disabled: false, label: tokens.eth.name, value: tokens.eth.name },
-    { disabled: false, label: tokens.bnb.name, value: tokens.bnb.name },
+    {
+      disabled: false,
+      label: tokens[network]["ERC-20"].name,
+      value: tokens[network]["ERC-20"].name,
+    },
+    {
+      disabled: false,
+      label: tokens[network]["BEP-20"].name,
+      value: tokens[network]["BEP-20"].name,
+    },
   ];
 
-  /** @type {String} */
-  let selected = tokens.eth.name;
+  /** @type {TokenNames} */
+  let selectedChain = tokens[network]["ERC-20"].name;
 
   /** @type {Boolean} */
   const migrationInProgress = false;
 
-  /** @type {Number} */
-  let connectedWalletBalance = 1;
+  /** @type {string} */
+  let connectedWalletBalance;
 
-  /** @type {Number | undefined} */
+  /** @type {Number} */
   let amount;
 
+  /** @type {import("svelte/store").Writable<Boolean>} */
+  const hasApprovedCoin = writable(false);
+
   /** @type {HTMLInputElement | null} */
   let amountInput;
 
   /** @type {Boolean} */
   let isMigrationInitialized = false;
 
-  /** @type {Number} */
-  const gasFee = 1;
-
-  /** @type {Boolean} */
-  let isMigrationBeingApproved = false;
-
-  /** @type {String}*/
-  const estimatedTime = "45 min";
-
   /** @type {number} */
   let screenWidth = window.innerWidth;
 
-  const { darkMode } = $settingsStore;
-  const minAmount = 1e-18;
+  /** @type {number} */
+  let migrationStep = 0;
 
-  $: ({ address, chainId, isConnected } = $account);
+  let isInputDisabled = false;
 
-  $: maxSpendable = connectedWalletBalance - gasFee;
-  $: isAmountValid =
-    typeof amount === "number"
-      ? amount >= minAmount && amount <= maxSpendable
-      : false;
-  $: ({ language } = $settingsStore);
-  $: duskFormatter = createCurrencyFormatter(language, "DUSK", 9);
+  function toggleDisabledAmount() {
+    isInputDisabled = !isInputDisabled;
+  }
+
+  const minAmount = 1e-9;
 
   /** @param {Number} id */
   async function handleSwitchChain(id) {
     try {
       await switchChain(wagmiConfig, { chainId: id });
     } catch (e) {
-      selected =
-        chainId === tokens.eth.chainId ? tokens.eth.name : tokens.bnb.name;
+      selectedChain =
+        chainId === tokens[network]["ERC-20"].chainId
+          ? tokens[network]["ERC-20"].name
+          : tokens[network]["BEP-20"].name;
     }
   }
```

Comment:
A refractor suggestion:

```
async function handleSwitchChain(chainId) {
  try {
    await switchChain(wagmiConfig, { chainId });
  } catch (error) {
    const { ["ERC-20"]: erc20, ["BEP-20"]: bep20 } = tokens[network];
    selectedChain = chainId === erc20.chainId ? erc20.name : bep20.name;
  }
}
```

If you find the solution with the destructuring assignment harder to read, I would at least suggest updating the `id` param name to `chainId` as it makes it easier to understand what the `id` (and also simplifies the `switchChain` call).